File: Programming_lab-06/Programming_lab-06/Customer.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Programming_lab_06
{
    struct Customer
    {
        private List<Shop> shop_in_mind_;

        //public Customer()
        //{
        //    shop_in_mind_ = new List<Shop>();
        //}

        public string check_string_;

        public List<Shop> Mind
        {
            get { return shop_in_mind_; }
        }

        public void rememberShop(Shop shop)
        {
            if (shop_in_mind_ == null) shop_in_mind_ = new List<Shop>();
            if (!(shop_in_mind_.Contains(shop)))
                shop_in_mind_.Add(shop);
        }

        public int buyGoods(string name, int quantity)
        {
            int n = shop_in_mind_.Count;
            for (int i = 0; quantity > 0 && i < n; i++)
                shop_in_mind_[i].sellBatchsOfGoods(name, ref quantity);
            return quantity;
        }
    }
}

File: Programming_lab-05_/src/demo/Customer.java

package demo;
import java.util.ArrayList;

public class Customer
{
	private ArrayList<Shop> shop_in_mind_;

      	public Customer()
      	{
        	shop_in_mind_ = new ArrayList<Shop>();
      	}

      	public void rememberShop(Shop shop)
      	{
          	if (!(shop_in_mind_.contains(shop)))
              		shop_in_mind_.add(shop);
      	}

      	public int buyGoods(String name, int quantity)
      	{
    	  	int[] q = new int[1];
    	  	q[0] = quantity;
          	int n = shop_in_mind_.size();
          	for (int i = 0; quantity > 0 && i < n; i++)
              		shop_in_mind_.get(i).sellBatchsOfGoods(name, q);
          	return q[0];
      	}
}